tests/test_pex_bootstrapper.py

```diff
@@ -28,3 +28,22 @@ def test_get_pex_info():
 
       # same when encoded
       assert pex_info.dump() == pex_info_2.dump()
+
+
+def test_find_compatible_interpreters():
+  ensure_python_interpreter('2.7.10')
+  ensure_python_interpreter('3.6.3')
+  pex_python_path = ':'.join([os.getcwd() + '/.pyenv_test/versions/2.7.10/bin/python2.7',
+                              os.getcwd() + '/.pyenv_test/versions/3.6.3/bin/python3.6'])
+
+  interpreters = find_compatible_interpreters(pex_python_path, ['>3'])
+  assert interpreters[0].binary == pex_python_path.split(':')[1]
+
+  interpreters = find_compatible_interpreters(pex_python_path, ['<3'])
+  assert interpreters[0].binary == pex_python_path.split(':')[0]
+
+  interpreters = find_compatible_interpreters(pex_python_path, ['<2'])
+  assert not interpreters
+
+  interpreters = find_compatible_interpreters(pex_python_path, ['>4'])
+  assert not interpreters
```

Comment:
how about adding a few more permutations here for more granular checking based around real world cases, like  a 2 vs 3 split or selecting '2.7.10' out of a PATH with '2.7.9', '2.7.10', and '2.7.11' etc?

Reply:
I believe this is already testing filtering based on a 2/3 split here, right? The current PPP has both 2 and 3 in it. I can certainly add testing for various other minor versions though.

Reply:
thinking more along the lines of fully exercising all 3 fields of `major.minor.patch` here vs just major versions.